File: src/main/java/io/saso/dash/services/github/GitHubService.java
package io.saso.dash.services.github;

import com.google.inject.Inject;
import io.netty.channel.ChannelHandlerContext;
import io.saso.dash.services.DBEntityProvider;
import io.saso.dash.services.Service;
import io.saso.dash.services.SubServiceFactory;
import org.kohsuke.github.GitHub;

public class GitHubService extends Service
{
    private final SubServiceFactory subServiceFactory;

    @Inject
    public GitHubService(SubServiceFactory subServiceFactory)
    {
        this.subServiceFactory = subServiceFactory;
    }

    @Override
    public void start(ChannelHandlerContext ctx, DBEntityProvider db)
            throws Exception
    {
        final GitHub gitHub =
                GitHub.connectUsingOAuth(db.authToken(this).getAccess());

        // notification
        registerSubPollable(
                subServiceFactory.createGitHubNotificationSubService(
                        this, gitHub));

        super.start(ctx, db);
    }

    @Override
    public int getPollInterval()
    {
        return 10;
    }

    @Override
    public String getProviderName()
    {
        return "github";
    }
}

File: src/main/kotlin/io/saso/dash/services/github/GitHubService.kt
package io.saso.dash.services.github

import com.google.inject.Inject
import io.netty.channel.ChannelHandlerContext
import io.saso.dash.services.DBEntityProvider
import io.saso.dash.services.Service
import io.saso.dash.services.SubServiceFactory
import io.saso.dash.util.logger
import org.kohsuke.github.GitHub

public class GitHubService
@Inject
constructor(private val subServiceFactory: SubServiceFactory) : Service()
{
    public override val pollInterval = -1

    override fun start(ctx: ChannelHandlerContext, db: DBEntityProvider)
    {
        val gitHub = GitHub.connectUsingOAuth(db.authToken("github").access)

        // notification
        registerSubPollable(
                subServiceFactory.createGitHubNotificationSubService(gitHub))

        super.start(ctx, db)
    }
}
